File: app/src/main/java/com/moh/mhealth/pages/PatientSurvey.java
package com.moh.mhealth.pages;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.NumberPicker;
import android.widget.TextView;

import com.moh.mhealth.Global_Helper;
import com.moh.mhealth.Header;
import com.moh.mhealth.Patient;
import com.moh.mhealth.R;

// Set village, time traveled, time waited
public class PatientSurvey extends AppCompatActivity implements Header {
    EditText villageField;

    NumberPicker distanceDaysPicker;
    NumberPicker distanceHoursPicker;

    NumberPicker timeHoursPicker;
    NumberPicker timeMinPicker;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_patient_survey);

        villageField = (EditText) findViewById(R.id.survey_village);

        distanceDaysPicker = (NumberPicker) findViewById(R.id.survey_distance_walked_days);
        distanceDaysPicker.setMinValue(0);
        distanceDaysPicker.setMaxValue(31);

        distanceHoursPicker = (NumberPicker) findViewById(R.id.survey_distance_walked_hours);
        distanceHoursPicker.setMinValue(0);
        distanceHoursPicker.setMaxValue(23);

        timeHoursPicker = (NumberPicker) findViewById(R.id.survey_time_waited_hr);
        timeHoursPicker.setMaxValue(48);
        timeHoursPicker.setMinValue(0);

        timeMinPicker = (NumberPicker) findViewById(R.id.survey_time_waited_min);
        timeMinPicker.setMaxValue(59);
        timeMinPicker.setMinValue(0);

        loadPatientData();

        Button saveBtn = (Button) findViewById(R.id.survey_next);

        saveBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                updatePatientData();

                startActivity(new Intent(getApplicationContext(), Global_Helper.nextDiag()));
            }
        });
    }

    private void updatePatientData() {

        Global_Helper.addPatientSurvey(villageField.getText().toString(),
                                       new int[] {distanceDaysPicker.getValue(),
                                                  distanceHoursPicker.getValue()},
                                       new int[] {timeHoursPicker.getValue(),
                                                  timeMinPicker.getValue()});
    }

    private void loadPatientData() {
        Patient patient = Global_Helper.getCurrentPatient();
        assert(patient != null);

        TextView patient_name = (TextView) findViewById(R.id.header_name);
        patient_name.setText(patient.getName());

        villageField.setText(patient.getVillage());

        int[] daysTraveled = patient.getDistance_traveled();
        distanceDaysPicker.setValue(daysTraveled[0]);
        distanceHoursPicker.setValue(daysTraveled[1]);

        int[] timeWaited = patient.getTime_waited();
        timeHoursPicker.setValue(timeWaited[0]);
        timeMinPicker.setValue(timeWaited[1]);
    }

    @Override
    public void cancel(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.endPatient()));
    }

    @Override
    public void moveBack(View view) {
        startActivity(new Intent(getApplicationContext(), Global_Helper.prevDiag()));
    }
}

File: app/src/main/java/com/moh/mhealth/pages/PatientSurvey.kt
package com.moh.mhealth.pages

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.NumberPicker
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.moh.mhealth.GlobalHelper
import com.moh.mhealth.Header
import com.moh.mhealth.R

// Set village, time traveled, time waited
class PatientSurvey : AppCompatActivity(), Header {
    private var villageField: EditText? = null
    private var distanceDaysPicker: NumberPicker? = null
    private var distanceHoursPicker: NumberPicker? = null
    private var timeHoursPicker: NumberPicker? = null
    private var timeMinPicker: NumberPicker? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_patient_survey)
        villageField = findViewById<View>(R.id.survey_village) as EditText
        distanceDaysPicker = findViewById<View>(R.id.survey_distance_walked_days) as NumberPicker
        distanceDaysPicker!!.minValue = 0
        distanceDaysPicker!!.maxValue = 31
        distanceHoursPicker = findViewById<View>(R.id.survey_distance_walked_hours) as NumberPicker
        distanceHoursPicker!!.minValue = 0
        distanceHoursPicker!!.maxValue = 23
        timeHoursPicker = findViewById<View>(R.id.survey_time_waited_hr) as NumberPicker
        timeHoursPicker!!.maxValue = 48
        timeHoursPicker!!.minValue = 0
        timeMinPicker = findViewById<View>(R.id.survey_time_waited_min) as NumberPicker
        timeMinPicker!!.maxValue = 59
        timeMinPicker!!.minValue = 0
        loadPatientData()
        val saveBtn = findViewById<View>(R.id.survey_next) as Button
        saveBtn.setOnClickListener {
            updatePatientData()
            startActivity(Intent(applicationContext, GlobalHelper.nextDiag()))
        }
    }

    private fun updatePatientData() {
        GlobalHelper.addPatientSurvey(
            villageField!!.text.toString(), intArrayOf(
                distanceDaysPicker!!.value,
                distanceHoursPicker!!.value
            ), intArrayOf(
                timeHoursPicker!!.value,
                timeMinPicker!!.value
            )
        )
    }

    private fun loadPatientData() {
        val patient = GlobalHelper.currentPatient!!
        val patientName = findViewById<View>(R.id.header_name) as TextView
        patientName.text = patient.name
        villageField!!.setText(patient.village)
        val daysTraveled = patient.distanceTraveled
        if (daysTraveled != null) {
            distanceDaysPicker!!.value = daysTraveled[0]
            distanceHoursPicker!!.value = daysTraveled[1]
        }
        val timeWaited = patient.timeWaited
        if (timeWaited != null) {
            timeHoursPicker!!.value = timeWaited   [0]
            timeMinPicker!!.value = timeWaited[1]
        }
    }

    override fun cancel(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.endPatient()))
    }

    override fun moveBack(view: View) {
        startActivity(Intent(applicationContext, GlobalHelper.prevDiag()))
    }
}